### ISSUE TITLE
When [Type] is specialized, it should map to base class [Type] column

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#2869]
Part of #2867. Depend on that #2868 is fixed.

Binding this:

```C#
public class Car {
  [Type] public CarModel Model;
}
public class Volvo : Car {
  [Type] public CarModel VolvoModel;
}
public class CarModel {}
```

should produce a schema with a *single column* for `Car.Model` and `Car.VolvoModel`. The later is thereby an implicit *synonym* to the former. Otherwise, it would be (at least theoretically) possible to assign a single object multiple type references.

**Comment by per-samuelsson** [comment]
This is also in line with efforts we did in `next` regarding implicit columns for `Entity` and non-Entity based database classes, and corresponds to the planned behaviour of 3.x, where they dynamic type column is part of every object.

**Comment by per-samuelsson** [comment]
I decided that I will make it so that classes deriving a class with an explicit type reference, need to specify their explicit type references (i.e. the specialized one), as a synonym. It's a bit more verbose, but I actually think there are more pros to this than cons:

* More explicit means easier to understand
* It *is* a synonym (it works just like one), so why shouldn't that be visible
* It derives the built-in constraints of synonyms, e.g. that the have to reference a compatible class (i.e. one *assignable* to the parent field).

**Comment by per-samuelsson** [comment]
This is now implemented, pushed as part of merge 3859b1b.

To have the model in the OP work, now define it like this:

```C#
public class Car {
  [Type] public CarModel Model;
}
public class Volvo : Car {
  [Type]
  [SynonymousTo("Model")]
  public CarModel VolvoModel;
}
public class CarModel {}
```
